src/presets/universal_deployer.cairo

```diff
@@ -0,0 +1,80 @@
+use starknet::ClassHash;
+use starknet::ContractAddress;
+
+#[starknet::interface]
+trait IUniversalDeployer<TState> {
+    fn deploy_contract(
+        ref self: TState,
+        class_hash: ClassHash,
+        salt: felt252,
+        unique: bool,
+        calldata: Span<felt252>
+    ) -> ContractAddress;
+}
+
+#[starknet::contract]
+mod UniversalDeployer {
+    use core::pedersen::pedersen;
+    use starknet::ClassHash;
+    use starknet::ContractAddress;
+    use starknet::SyscallResultTrait;
+    use starknet::get_caller_address;
+
+    #[storage]
+    struct Storage {}
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        ContractDeployed: ContractDeployed
+    }
+
+    #[derive(Drop, starknet::Event)]
+    struct ContractDeployed {
+        address: ContractAddress,
+        deployer: ContractAddress,
+        unique: bool,
+        class_hash: ClassHash,
+        calldata: Span<felt252>,
+        salt: felt252,
+    }
+
+    #[abi(embed_v0)]
+    impl UniversalDeployerImpl of super::IUniversalDeployer<ContractState> {
+        fn deploy_contract(
+            ref self: ContractState,
+            class_hash: ClassHash,
+            salt: felt252,
+            unique: bool,
+            calldata: Span<felt252>
+        ) -> ContractAddress {
+            let deployer: ContractAddress = get_caller_address();
+
+            // Defaults for non-unique deployment
+            let mut _salt: felt252 = salt;
+            let mut from_zero: bool = true;
+
+            if unique {
+                _salt = pedersen(deployer.into(), salt);
```

Comment:
Should we consider using poseidon since is much cheaper? Maybe a UDC_v2?

Reply:
Huge +1 to UDC_v2 with poseidon

Reply:
I think we may need to be explicit in the name regarding the version. If people are using UDC already to predict addresses before deployment, getting confused about whether it uses pedersen or poseidon could be problematic, since the addresses would be different.

Reply:
Should we update to Poseidon then on this PR? And should we add documentation about it and versions? In the starknet docsite we are not for example mentioning which hash algorithm is used, or how to precompute the addresses, and I think that's worth to mention.

Reply:
Made this an issue: #950